fips.Dockerfile

```diff
@@ -0,0 +1,134 @@
+# check=skip=RedundantTargetPlatform
+# setup build image
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.24.2-fips-bullseye@sha256:28ab4742d3b5feb0b3c7450629b4e105128d8709dd6ca22898472ab302140c37 AS operator-build
+
+ENV GOEXPERIMENT=systemcrypto
+
+WORKDIR /app
+
+ARG DEBUG_TOOLS
+RUN if [ "$DEBUG_TOOLS" = "true" ]; then \
+      GOBIN=/app/build/_output/bin go install github.com/go-delve/delve/cmd/dlv@latest; \
+    fi
+
+COPY go.mod go.sum ./
+RUN go mod download -x
+
+COPY pkg ./pkg
+COPY cmd ./cmd
+
+ARG GO_LINKER_ARGS
+ARG GO_BUILD_TAGS
+ARG TARGETARCH
+ARG TARGETOS
+
+RUN --mount=type=cache,target="/root/.cache/go-build" \
+    --mount=type=cache,target="/go/pkg" \
+    CGO_ENABLED=1 GOFIPS=1 GOOS=$TARGETOS GOARCH=$TARGETARCH \
+    go build -tags "${GO_BUILD_TAGS}" -trimpath -ldflags="${GO_LINKER_ARGS}" \
+    -o ./build/_output/bin/dynatrace-operator ./cmd/
+
+# ---------------- Install Packages in Final Image -----------------------
+# platform is required, otherwise the copy command will copy the wrong architecture files, don't trust GitHub Actions linting warnings
+FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9-micro:9.5-1744118077@sha256:dca8bc186bb579f36414c6ad28f1dbeda33e5cf0bd5fc1c51430cc578e25f819 AS base
+FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9:9.5-1744101466@sha256:ea57285741f007e83f2ee20423c20b0cbcce0b59cc3da027c671692cc7efe4dd AS dependency
+RUN mkdir -p /tmp/rootfs-dependency
+COPY --from=base / /tmp/rootfs-dependency
+RUN dnf install --installroot /tmp/rootfs-dependency \
+      util-linux-core tar \
+      --releasever 9 \
+      --setopt install_weak_deps=false \
+      --nodocs -y \
+ && dnf --installroot /tmp/rootfs-dependency clean all \
+ && rm -rf \
+      /tmp/rootfs-dependency/var/cache/* \
+      /tmp/rootfs-dependency/var/log/dnf* \
+      /tmp/rootfs-dependency/var/log/yum.*
+
+# Build and Install OpenSSL
+# Version must be FIPS certified, details https://openssl-library.org/source/
+ENV OPENSSL_BUILD_VERSION="3.1.2"
```

Comment:
more q for future maintaince: do we plan to update it using renovate later?

Reply:
AFAIK not really possible, I have failed to find any (reasonable and) automatable way to determine which is the latest OpenSSL version that is FIPS verified.

Reply:
what about to make it as ARG and default ENV value so easy to update later? we can iterate on it later.

Reply:
I would leave it as is